tests/upgrades/test_client.py

```diff
@@ -149,42 +107,30 @@ def test_pre_scenario_preclient_package_installation(
             2. The new repo is enabled on the content host.
 
         """
-        rhel7_client = dockerize(
-            ak_name=module_ak.name, distro='rhel7', org_label=default_org.label
-        )
-        client_container_id = list(rhel7_client.values())[0]
-        # Wait 60 seconds as script in /tmp takes up to 2 min inc the repo sync time
-        time.sleep(60)
-        # Use yum install again in case script was not yet finished
-        installed_package = execute(
-            docker_execute_command,
-            client_container_id,
-            'yum -y install katello-agent',
-            host=docker_vm,
-        )[docker_vm]
-        # Assert gofer was installed after yum completes
-        installed_package = execute(
-            docker_execute_command,
-            client_container_id,
-            'rpm -q gofer',
-            host=docker_vm,
-        )[docker_vm]
-        assert 'package gofer is not installed' not in installed_package
+        rhel_client = katello_agent_client['client']
+        rhid = rhel_client.execute('subscription-manager identity')
+        assert module_org.name in rhid.stdout
+        result = rhel_client.execute('rpm -qa | grep gofer')
+        assert result.status == 0
+        assert 'package gofer is not installed' not in result.stdout
         # Run goferd on client as its docker container
-        kwargs = {'async': True, 'host': docker_vm}
-        execute(docker_execute_command, client_container_id, 'goferd -f', **kwargs)
+        rhel_client.execute("goferd -f")
```

Comment:
This isnt seems to be async operation, We need to make it async so that we move on to the next part and wait for 30 seconds.

Reply:
Do you mean that we need to add the sleep here for 30 secs?

Reply:
We need to make sure if `goferd -f` is an asynchronous operation or synchronous operation. If its synchronous, then we have to make `rhel_client.execute`  to run that in background else we wont fwd to the next step and then wait for 30 seconds assuming it dones with something we are waiting for here.

Reply:
Is this client intended to be a container or a traditional VM?
We have some established ways of starting goferd in both. see:
https://github.com/SatelliteQE/robottelo/blob/master/robottelo/hosts.py#L417